packages/dds/tree/src/test/feature-libraries/default-schema/schemaChecker.spec.ts

```diff
@@ -0,0 +1,722 @@
+				);
+
+				assert.equal(
+					isNodeInSchema(mapNode_oneNumber, {
+						schema: emptySchemaCollection,
+						policy: schemaPolicy,
+					}),
+					SchemaValidationErrors.Node_MissingSchema,
+				);
+			});
+
+			it(`not in schema due to missing FieldKind entry in schemaPolicy`, () => {
+				// Schema for a map node whose fields are required and must contain a number.
+				const mapNodeSchema: TreeNodeStoredSchema = new MapNodeStoredSchema(
+					getFieldSchema(FieldKinds.required, [numberNode.type]),
+				);
+
+				// numberNode.type is not in the schema collection
+				const mapNode_oneNumber = getMapNode(
+					"myNumberMapNode",
+					new Map([[brand("prop1"), [numberNode]]]),
+				);
+
+				const schemaCollection: StoredSchemaCollection = {
+					nodeSchema: new Map([
+						[numberNode.type, new LeafNodeStoredSchema(ValueSchema.Number)],
+						[mapNode_oneNumber.type, mapNodeSchema],
+					]),
+				};
+
+				assert.equal(
+					isNodeInSchema(mapNode_oneNumber, {
+						schema: schemaCollection,
+						policy: emptySchemaPolicy,
+					}),
+					SchemaValidationErrors.Field_KindNotInSchemaPolicy,
+				);
+			});
+
+			it(`not in schema if nodes are not allowed by field`, () => {
+				const fieldSchema_requiredNumberNode = getFieldSchema(FieldKinds.required, [
+					numberNode.type,
+				]);
+				const fieldSchema_requiredStringNode = getFieldSchema(FieldKinds.required, [
+					stringNode.type,
+				]);
+				const mapNodeSchema: TreeNodeStoredSchema = new MapNodeStoredSchema(
+					fieldSchema_requiredNumberNode,
+				);
+				const mapNode = getMapNode(
+					"myNumberMapNode",
+					new Map([[brand("prop1"), [numberNode]]]),
+				);
+
+				const schemaCollection = {
+					nodeSchema: new Map([
+						[numberNode.type, new LeafNodeStoredSchema(ValueSchema.Number)],
+						[mapNode.type, mapNodeSchema],
+					]),
+				};
+				const schemaPolicy = {
+					fieldKinds: new Map([
+						[fieldSchema_requiredNumberNode.kind, FieldKinds.required],
+						[fieldSchema_requiredStringNode.kind, FieldKinds.required],
+					]),
+				};
+
+				// In schema with one number node
+				assert.equal(
+					isNodeInSchema(mapNode, { schema: schemaCollection, policy: schemaPolicy }),
+					SchemaValidationErrors.NoError,
+				);
+
+				// Not in schema after adding a string node
+				mapNode.fields.set(brand("prop2"), [stringNode]);
+				assert.equal(
+					isNodeInSchema(mapNode, { schema: schemaCollection, policy: schemaPolicy }),
+					SchemaValidationErrors.Field_NodeTypeNotAllowed,
+				);
+
+				// mapNode.fields.set(brand("prop2"), []);
+			});
+		});
+
+		describe("ObjectNodeStoredSchema", () => {
+			const numberNode = getValueNode("myNumberNode", 1);
+			const stringNode = getValueNode("myStringNode", "string");
+			const booleanNode = getValueNode("myBooleanNode", false);
+			const nullNode = getValueNode("myNullNode", null);
+			const fluidHandleNode = getValueNode("myFluidHandleNode", new TestFluidHandle());
```

Comment:
Object nodes don't treat different value types any different, so there really should be a need to have one of each for testing. Also it seems like most of what you are testing for object nodes is actually field validation, which has its own tests.

I think object node tests should prabably focus on the cases that are actually specific to object node handling. Maybe:

1. Extra required field on schema.
2. Extra field on node.
3. value on node. (I don't think you test or handle this case correctly)
4. valid with required field
5. valid with empty optional field

Reply:
I think its worth noting that testing way more than you need to makes it hard for someone reading the tests to notice something is missed. Minimal well orthogonalized unit tests are easier to see if they cover everything. I think you could get by with less than a third the test code here and still cover everything

Reply:
True on the redundancy with field validation, I got rid of some of those, and simplified the ones for valid scenarios.

I guess I'd say I was leaning towards testing the contract of the function more than its implementation (like mentioned in another comment, if `isNodeInSchema` stopped relying on `isFieldInSchema` for some reason, I'd still want to guarantee that it will fail in all the appropriate scenarios). But I agree it didn't feel like an obviously ideal set of tests, and that overall it feels excessive rather than lacking, so happy to reduce/simplify them.